sdk/core/azure-core-tracing-opentelemetry/tests/test_tracing_implementations.py

```diff
@@ -271,6 +271,31 @@ def test_links_with_attribute(self, tracer):
                 assert "attribute2" in link.attributes
                 assert link.attributes == attributes
 
+    def test_parent_context(self, tracer):
+        with tracer.start_as_current_span("Root"):
+            parent_header = {"traceparent": "00-2578531519ed94423ceae67588eff2c9-231ebdc614cb9ddd-01"}
+            span = OpenTelemetrySpan(name="test_span", context=parent_header)
+            assert span.span_instance.context.trace_id == int("2578531519ed94423ceae67588eff2c9", 16)
+            assert span.span_instance.parent.span_id == int("231ebdc614cb9ddd", 16)
+
+    def test_empty_parent_context(self, tracer):
+        with tracer.start_as_current_span("Root") as root:
+            span = OpenTelemetrySpan(name="test_span", context={})
+            # Parent of span should be None if context is empty (orphaned span).
+            assert span.span_instance.parent is None
```

Comment:
@lmolkova, would appreciate your input in this specific case since I plan on using this for EventHubs process spans...

Based on [the gist](https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#processing-messages), in the case of processing a single event, the process span should be a direct child of the message context. If no context is available (i.e., we pass in empty context), then the parent is expected to be None even if the span is created in the scope of another span? That's the current behavior here, but wanted to verify.

Reply:
great catch. I slightly updated the gist and removed sentence on orphaned spans.

https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92/revisions

There is no need for special handling of no-context-in-the-message case. If there is a current span (which should not be the case), it can be used. Having an ambient span there would probably indicate some issue in user code.

The assumption is that users set up the processor at startup time and don't have a chance to start span in the execution context in which processor callback is executed.

So feel free to simplify this code and remove the edge ase